src/globus_sdk/_globus_sdk_flake8.py

```diff
@@ -124,16 +126,35 @@ def visit_Assign(self, node: ast.Assign) -> None:
     #     type_ignores=[])
     #
     def visit_Call(self, node: ast.Call) -> None:
-        # if it's not a call to 'x.info(...)', ignore
-        if not (isinstance(node.func, ast.Attribute) and node.func.attr == "info"):
-            self.generic_visit(node)
-            return
-        func_node: ast.Attribute = node.func
-
-        # if the function was not a method of something named "log", ignore
-        if not (isinstance(func_node.value, ast.Name) and func_node.value.id == "log"):
+        # +---------------------------------------------+
+        # | check SDK003 | `isinstance(x, MissingType)` |
+        # +---------------------------------------------+
+        if (
+            # an `isinstance()` call with two arguments
+            # (really just means it's a valid call)
+            isinstance(node.func, ast.Name)
+            and node.func.id == "isinstance"
+            and len(args := node.args) == 2
+            # where the second argument is a name node,
+            # not a tuple or other expression
+            and isinstance(rhs := args[1], ast.Name)
+            # and the name used is 'MissingType'
+            and rhs.id == "MissingType"
+        ):
+            self._record(node, "SDK003")
+
+        # +--------------------------------+
+        # | check SDK002 | `log.info(...)` |
+        # +--------------------------------+
+        elif (
+            # the function call is of the form 'OBJ.info(...)'
+            isinstance(func_node := node.func, ast.Attribute)
+            and func_node.attr == "info"
+            # and, more specifically, the object is named 'log', so
+            # it's `log.info(...)
+            and isinstance(func_node.value, ast.Name)
+            and func_node.value.id == "log"
+        ):
+            self._record(node, "SDK002")
+        else:
```

Comment:
I'm tickled that it's possible to express the check this way. It's clever, but... I think we should turn these into functions for readability. I think the necessity of the comments gives away that this is rather tricky to read as it's currently expressed. I think we could express the same logic more plainly.